airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/client/client.py

```diff
@@ -0,0 +1,312 @@
+        'age', 'gender', 'country', 'placement', 'region', 'dma'
+    ]
+
+    action_breakdowns = ALL_ACTION_BREAKDOWNS
+    level = 'ad'
+    action_attribution_windows = ALL_ACTION_ATTRIBUTION_WINDOWS
+    time_increment = 1
+    buffer_days = 28
+
+    def __init__(self, api, start_date, breakdowns=None):
+        super().__init__(api=api)
+        self.start_date = start_date
+        self.breakdowns = breakdowns
+
+    def list(self, fields: Sequence[str] = None) -> Generator[dict]:
+        for params in self._params():
+            for obj in self._get_insights(params):
+                rec = obj.export_all_data()
+                yield rec
+
+    def _params(self, fields: Sequence[str] = None) -> Generator[dict]:
+        buffered_start_date = self.start_date.subtract(days=self.buffer_days)
+        end_date = datetime.now()
+
+        fields = list(set(fields) - set(self.INVALID_INSIGHT_FIELDS))
+
+        # Some automatic fields (primary-keys) cannot be used as 'fields' query params.
+        while buffered_start_date <= end_date:
+            yield {
+                'level': self.level,
+                'action_breakdowns': self.action_breakdowns,
+                'breakdowns': self.breakdowns,
+                'limit': self.result_return_limit,
+                'fields': fields,
+                'time_increment': self.time_increment,
+                'action_attribution_windows': self.action_attribution_windows,
+                'time_ranges': [{'since': buffered_start_date.to_date_string(),
+                                 'until': buffered_start_date.to_date_string()}]
+            }
+            buffered_start_date = buffered_start_date.add(days=1)
+
+    @retry_pattern(backoff.expo, FacebookRequestError, max_tries=5, factor=5)
+    def _get_insights(self, params):
+        return self._api.account.get_insights(params=params)
+
+
+class Client(BaseClient):
+    def __init__(self, account_id: str, access_token: str, start_date: str):
+        super().__init__()
+        self._api = FacebookAdsApi.init(access_token=access_token)
+        self._account = self._find_account(account_id)
+        self._start_date = isoparse(start_date)
+
+    def stream__campaigns(self, fields):
+        yield from CampaignAPI(self).list(fields)
+
+    def stream__ad_sets(self, fields):
+        yield from AdSetsAPI(self).list(fields)
+
+    def stream__ads(self, fields):
+        yield from AdsAPI(self).list(fields)
+
+    def stream__adcreatives(self, fields):
+        yield from AdCreativeAPI(self).list(fields)
+
+    def stream__ads_insight(self, fields):
+        client = AdsInsightAPI(self, start_date=self._start_date)
+        yield from client.list(fields)
+
+    def stream__ads_insight_age_and_gender(self, fields):
+        breakdowns = ['age', 'gender']
+        client = AdsInsightAPI(self, start_date=self._start_date, breakdowns=breakdowns)
+        yield from client.list(fields)
```

Comment:
What if implement this method (and the below ones) with forwarding call to `stream__ads_insight`:
```
yield from self.stream__ads_insight(fields, breakdowns=['age', 'gender'])
```
`stream__ads_insights` signature to be modified though by adding `**kwargs`